src/action/base/add_user_to_group.rs

```diff
@@ -0,0 +1,284 @@
+
+/**
+Create an operating system level user in the given group
+*/
+#[derive(Debug, serde::Deserialize, serde::Serialize, Clone)]
+pub struct AddUserToGroup {
+    name: String,
+    uid: u32,
+    groupname: String,
+    gid: u32,
+}
+
+impl AddUserToGroup {
+    #[tracing::instrument(level = "debug", skip_all)]
+    pub async fn plan(
+        name: String,
+        uid: u32,
+        groupname: String,
+        gid: u32,
+    ) -> Result<StatefulAction<Self>, ActionError> {
+        let this = Self {
+            name: name.clone(),
+            uid,
+            groupname,
+            gid,
+        };
+        // Ensure user does not exists
+        if let Some(user) = User::from_name(name.as_str())
+            .map_err(|e| ActionError::GettingUserId(name.clone(), e))?
+        {
+            if user.uid.as_raw() != uid {
+                return Err(ActionError::UserUidMismatch(
+                    name.clone(),
+                    user.uid.as_raw(),
+                    uid,
+                ));
+            }
+
+            if user.gid.as_raw() != gid {
+                return Err(ActionError::UserGidMismatch(
+                    name.clone(),
+                    user.gid.as_raw(),
+                    gid,
+                ));
+            }
+
+            // See if group membership needs to be done
+            match target_lexicon::OperatingSystem::host() {
+                OperatingSystem::MacOSX {
+                    major: _,
+                    minor: _,
+                    patch: _,
+                }
+                | OperatingSystem::Darwin => {
+                    let mut command = Command::new("/usr/sbin/dseditgroup");
+                    command.process_group(0);
+                    command.args(["-o", "checkmember", "-m"]);
+                    command.arg(&this.name);
+                    command.arg(&this.groupname);
+                    command.stdout(Stdio::piped());
+                    command.stderr(Stdio::piped());
+                    let command_str = format!("{:?}", command.as_std());
+                    tracing::trace!("Executing `{command_str}`");
+                    let output = command.output().await.map_err(ActionError::Command)?;
+                    match output.status.code() {
+                        Some(0) => {
+                            // yes {user} is a member of {groupname}
+                            // Since the user exists, and is already a member of the group, we have truly nothing to do here
+                            tracing::debug!(
+                                "Adding user `{}` to group `{}` already complete",
+                                this.name,
+                                this.groupname
+                            );
+                            return Ok(StatefulAction::completed(this));
+                        },
+                        Some(64) => {
+                            // Group not found
```

Comment:
Is there a snippet that we can paste here or docs we can link to that tells us `64` means this?

Reply:
This was the intention of the comment, but I'll try to dig something up!

Reply:
Not really fussed, was just hoping there was some prevalent documentation / some salient wording we could refer to. If that's not the case, the current note is fine.